components/raftstore/src/store/peer.rs

```diff
@@ -3735,9 +3811,21 @@ where
                         cp
                     ));
                 }
-                (ConfChangeType::RemoveNode, _)
-                | (ConfChangeType::AddNode, PeerRole::Voter)
-                | (ConfChangeType::AddLearnerNode, PeerRole::Learner) => {}
+                (ConfChangeType::RemoveNode, _) => {}
+                (ConfChangeType::AddNode, PeerRole::Voter)
+                | (ConfChangeType::AddLearnerNode, PeerRole::Learner) => {
+                    if peer.get_id() == self.peer_id()
```

Comment:
When the witness is converted to non-witness and the non-witness is waiting for apply snapshot, the operator of pd may push witness -> non-witness conf change again, should we filter out this conf change here? Otherwise, the apply thread will print an error message, may cause misjudgment?

Reply:
Yes, I will add a check to ignore duplicated operators.

Reply:
yes :)

Reply:
I think maybe leader should put future non-witness into `wait_data_peers`, as the non-witness peer may `request_snapshot` before leader finish apply `witness->non-witness` conf change. Then leader will send a witness snapshot to the non-witness peer.